src/modules/BuilderPlace/context/BuilderPlaceContext.tsx

```diff
@@ -1,34 +1,79 @@
-import { createContext, ReactNode, useContext, useEffect, useMemo, useState } from 'react';
+import { createContext, ReactNode, useEffect, useState } from 'react';
 import { IBuilderPlace } from '../types';
 import { useAccount } from 'wagmi';
+import { getUserById } from '../../../queries/users';
+import { toast } from 'react-toastify';
+import { useChainId } from '../../../hooks/useChainId';
 
 const BuilderPlaceContext = createContext<{
   builderPlace?: IBuilderPlace;
+  isBuilderPlaceCollaborator: boolean;
   isBuilderPlaceOwner: boolean;
 }>({
   builderPlace: undefined,
+  isBuilderPlaceCollaborator: false,
   isBuilderPlaceOwner: false,
 });
 
 const BuilderPlaceProvider = ({ data, children }: { data: IBuilderPlace; children: ReactNode }) => {
   const account = useAccount();
+  const chainId = useChainId();
   const [builderPlace, setBuilderPlace] = useState<IBuilderPlace | undefined>();
-  const [isBuilderPlaceOwner, setIsBuilderPlaceOwner] = useState<boolean>(false);
+  const [isBuilderPlaceCollaborator, setIsBuilderPlaceCollaborator] = useState<boolean>(false);
+  const [isBuilderPlaceOwner, setIsBuilderPlaceOwner] = useState(false);
+
+  const fetchBuilderPlaceOwner = async () => {
+    if (!data?.ownerTalentLayerId) {
+      return;
+    }
+
+    try {
+      const response = await getUserById(chainId, data.ownerTalentLayerId);
```

Comment:
it's seems not for more optimal way to calculate isUserBuilderPlaceOwner. We already have in the TalentLayer context the current user. So better to not make another API call and just get the info from there.

Reply:
I put this call here since I need the infos of another user (BuilderPlace's owner) and not the connected user.

This will be removed when the Owner's address is set in the DataBase.